service/handlers_substore_test.go

```diff
@@ -0,0 +1,221 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017-2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package service
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/CanonicalLtd/serial-vault/config"
+	"github.com/CanonicalLtd/serial-vault/datastore"
+	check "gopkg.in/check.v1"
+)
+
+func TestSubstoreSuite(t *testing.T) { check.TestingT(t) }
+
+type SubstoreSuite struct{}
+
+type SubstoreTest struct {
+	Method      string
+	URL         string
+	Data        []byte
+	Code        int
+	Type        string
+	Permissions int
+	EnableAuth  bool
+	Success     bool
+	Stores      int
+}
+
+var _ = check.Suite(&SubstoreSuite{})
+
+func (s *SubstoreSuite) SetUpTest(c *check.C) {
+	// Mock the database
+	config := config.Settings{KeyStoreType: "filesystem", KeyStorePath: "../keystore", JwtSecret: "SomeTestSecretValue"}
+	datastore.Environ = &datastore.Env{DB: &datastore.MockDB{}, Config: config}
+	datastore.OpenKeyStore(config)
+}
+
+func (s *SubstoreSuite) sendRequest(method, url string, data io.Reader, permissions int, c *check.C) *httptest.ResponseRecorder {
```

Comment:
Actually you don't need sendRequest, parseSubstoresResponse or parseBooleanResponse to have a pointer receiver to the suite, they are not using anything from the Suite